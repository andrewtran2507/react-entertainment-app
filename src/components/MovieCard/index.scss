#storybook-root {
  .movie-card {
    width: 299px;
  }
}
#story--components-moviecard--un-loading-inner,
#story--components-moviecard--on-loading-inner,
#story--components-moviecard--on-loading--primary-inner {
  .movie-card {
    padding: 10px 368px;
  }
}
.movie-card {
  padding: 10px;
  .skeleton-img {
    margin-bottom: 10px;
    .ant-skeleton {
      width: 100%;
      .ant-skeleton-image {
        width: unset;
        height: 200px;
      }
    }
  }
  .ant-card-cover {
    img {
      width: inherit;
    }
  }
}
