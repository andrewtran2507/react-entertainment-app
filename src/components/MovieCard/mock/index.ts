import {
  IMovieState,
  ZMovieItemModel,
  ZMovieResponseModel,
} from 'pages/Home/Redux/types';
import { IErrorResponse } from 'utils/constant';

export const movieItem1 = {
  // API wrong data to check zod return wrong
  results: [
    {
      adult: false,
      backdrop_path: [],
      genre_ids: [28, 80, 53],
      id: 385687,
      original_language: 'en',
      original_title: 'Fast X',
      overview:
        "Over many missions and against impossible odds, Dom Toretto and his family have outsmarted, out-nerved and outdriven every foe in their path. Now, they confront the most lethal opponent they've ever faced: A terrifying threat emerging from the shadows of the past who's fueled by blood revenge, and who is determined to shatter this family and destroy everything—and everyone—that Dom loves, forever.",
      popularity: 1434.884,
      poster_path: 12,
      release_date: '2023-05-17',
      a: 'Fast X',
      b: [1231312],
      video: false,
      vote_average: 'sdsdsds',
      vote_count: 4218,
    },
  ],
  total_pages: 12,
  total_results: 12,
  page: 1,
};

export const movieItem: ZMovieItemModel = {
  adult: false,
  backdrop_path: '/4XM8DUTQb3lhLemJC51Jx4a2EuA.jpg',
  genre_ids: [28, 80, 53],
  id: 385687,
  original_language: 'en',
  original_title: 'Fast X',
  overview:
    "Over many missions and against impossible odds, Dom Toretto and his family have outsmarted, out-nerved and outdriven every foe in their path. Now, they confront the most lethal opponent they've ever faced: A terrifying threat emerging from the shadows of the past who's fueled by blood revenge, and who is determined to shatter this family and destroy everything—and everyone—that Dom loves, forever.",
  popularity: 1434.884,
  poster_path: '/fiVW06jE7z9YnO4trhaMEdclSiC.jpg',
  release_date: '2023-05-17',
  title: 'Fast X Example film',
  video: false,
  vote_average: 7.2,
  vote_count: 4218,
};

const errorResItem: IErrorResponse = {
  status_code: 7,
  status_message: 'Invalid API key: You must be granted a valid key.',
  success: false,
};

export const errorItem: IMovieState = {
  loading: false,
  data: null,
  error: errorResItem,
};

export const movieItem2: ZMovieResponseModel = {
  page: 1,
  results: [
    {
      adult: false,
      backdrop_path: '/fm6KqXpk3M2HVveHwCrBSSBaO0V.jpg',
      genre_ids: [18, 36],
      id: 872585,
      original_language: 'en',
      original_title: 'Oppenheimer',
      overview:
        "The story of J. Robert Oppenheimer's role in the development of the atomic bomb during World War II.",
      popularity: 2590.24,
      poster_path: '/8Gxv8gSFCU0XGDykEGv7zR1n2ua.jpg',
      release_date: '2023-07-19',
      title: 'Oppenheimer',
      video: false,
      vote_average: 8.2,
      vote_count: 4588,
    },
    {
      adult: false,
      backdrop_path: '/t5zCBSB5xMDKcDqe91qahCOUYVV.jpg',
      genre_ids: [27, 9648],
      id: 507089,
      original_language: 'en',
      original_title: "Five Nights at Freddy's",
      overview:
        "Recently fired and desperate for work, a troubled young man named Mike agrees to take a position as a night security guard at an abandoned theme restaurant: Freddy Fazbear's Pizzeria. But he soon discovers that nothing at Freddy's is what it seems.",
      popularity: 1617.893,
      poster_path: '/A4j8S6moJS2zNtRR8oWF08gRnL5.jpg',
      release_date: '2023-10-25',
      title: "Five Nights at Freddy's",
      video: false,
      vote_average: 8,
      vote_count: 2300,
    },
    {
      adult: false,
      backdrop_path: '/kjQBrc00fB2RjHZB3PGR4w9ibpz.jpg',
      genre_ids: [878, 28, 53],
      id: 670292,
      original_language: 'en',
      original_title: 'The Creator',
      overview:
        'Amid a future war between the human race and the forces of artificial intelligence, a hardened ex-special forces agent grieving the disappearance of his wife, is recruited to hunt down and kill the Creator, the elusive architect of advanced AI who has developed a mysterious weapon with the power to end the war—and mankind itself.',
      popularity: 1595.817,
      poster_path: '/vBZ0qvaRxqEhZwl6LWmruJqWE8Z.jpg',
      release_date: '2023-09-27',
      title: 'The Creator',
      video: false,
      vote_average: 7.2,
      vote_count: 786,
    },
    {
      adult: false,
      backdrop_path: '/628Dep6AxEtDxjZoGP78TsOxYbK.jpg',
      genre_ids: [28, 53],
      id: 575264,
      original_language: 'en',
      original_title: 'Mission: Impossible - Dead Reckoning Part One',
      overview:
        "Ethan Hunt and his IMF team embark on their most dangerous mission yet: To track down a terrifying new weapon that threatens all of humanity before it falls into the wrong hands. With control of the future and the world's fate at stake and dark forces from Ethan's past closing in, a deadly race around the globe begins. Confronted by a mysterious, all-powerful enemy, Ethan must consider that nothing can matter more than his mission—not even the lives of those he cares about most.",
      popularity: 1044.199,
      poster_path: '/NNxYkU70HPurnNCSiCjYAmacwm.jpg',
      release_date: '2023-07-08',
      title: 'Mission: Impossible - Dead Reckoning Part One',
      video: false,
      vote_average: 7.6,
      vote_count: 2454,
    },
    {
      adult: false,
      backdrop_path: '/rMvPXy8PUjj1o8o1pzgQbdNCsvj.jpg',
      genre_ids: [28, 12, 53],
      id: 299054,
      original_language: 'en',
      original_title: 'Expend4bles',
      overview:
        'Armed with every weapon they can get their hands on and the skills to use them, The Expendables are the world’s last line of defense and the team that gets called when all other options are off the table. But new team members with new styles and tactics are going to give “new blood” a whole new meaning.',
      popularity: 1108.183,
      poster_path: '/iwsMu0ehRPbtaSxqiaUDQB9qMWT.jpg',
      release_date: '2023-09-15',
      title: 'Expend4bles',
      video: false,
      vote_average: 6.4,
      vote_count: 741,
    },
    {
      adult: false,
      backdrop_path: '/5a4JdoFwll5DRtKMe7JLuGQ9yJm.jpg',
      genre_ids: [28, 12, 878],
      id: 695721,
      original_language: 'en',
      original_title: 'The Hunger Games: The Ballad of Songbirds & Snakes',
      overview:
        '64 years before he becomes the tyrannical president of Panem, Coriolanus Snow sees a chance for a change in fortunes when he mentors Lucy Gray Baird, the female tribute from District 12.',
      popularity: 734.432,
      poster_path: '/mBaXZ95R2OxueZhvQbcEWy2DqyO.jpg',
      release_date: '2023-11-15',
      title: 'The Hunger Games: The Ballad of Songbirds & Snakes',
      video: false,
      vote_average: 7.4,
      vote_count: 146,
    },
    {
      adult: false,
      backdrop_path: '/4XM8DUTQb3lhLemJC51Jx4a2EuA.jpg',
      genre_ids: [28, 80, 53],
      id: 385687,
      original_language: 'en',
      original_title: 'Fast X',
      overview:
        "Over many missions and against impossible odds, Dom Toretto and his family have outsmarted, out-nerved and outdriven every foe in their path. Now, they confront the most lethal opponent they've ever faced: A terrifying threat emerging from the shadows of the past who's fueled by blood revenge, and who is determined to shatter this family and destroy everything—and everyone—that Dom loves, forever.",
      popularity: 770.248,
      poster_path: '/fiVW06jE7z9YnO4trhaMEdclSiC.jpg',
      release_date: '2023-05-17',
      title: 'Fast X',
      video: false,
      vote_average: 7.2,
      vote_count: 4283,
    },
    {
      adult: false,
      backdrop_path: '/feSiISwgEpVzR1v3zv2n2AU4ANJ.jpg',
      genre_ids: [28, 12, 878],
      id: 609681,
      original_language: 'en',
      original_title: 'The Marvels',
      overview:
        'Carol Danvers, aka Captain Marvel, has reclaimed her identity from the tyrannical Kree and taken revenge on the Supreme Intelligence. But unintended consequences see Carol shouldering the burden of a destabilized universe. When her duties send her to an anomalous wormhole linked to a Kree revolutionary, her powers become entangled with that of Jersey City super-fan Kamala Khan, aka Ms. Marvel, and Carol’s estranged niece, now S.A.B.E.R. astronaut Captain Monica Rambeau. Together, this unlikely trio must team up and learn to work in concert to save the universe.',
      popularity: 633.789,
      poster_path: '/Ag3D9qXjhJ2FUkrlJ0Cv1pgxqYQ.jpg',
      release_date: '2023-11-08',
      title: 'The Marvels',
      video: false,
      vote_average: 6.5,
      vote_count: 430,
    },
    {
      adult: false,
      backdrop_path: '/tC78Pck2YCsUAtEdZwuHYUFYtOj.jpg',
      genre_ids: [28, 53, 80],
      id: 926393,
      original_language: 'en',
      original_title: 'The Equalizer 3',
      overview:
        "Robert McCall finds himself at home in Southern Italy but he discovers his friends are under the control of local crime bosses. As events turn deadly, McCall knows what he has to do: become his friends' protector by taking on the mafia.",
      popularity: 741.736,
      poster_path: '/b0Ej6fnXAP8fK75hlyi2jKqdhHz.jpg',
      release_date: '2023-08-30',
      title: 'The Equalizer 3',
      video: false,
      vote_average: 7.4,
      vote_count: 1646,
    },
    {
      adult: false,
      backdrop_path: '/3H9NA1KWEQN0ItL3Wl3SFZYP6yV.jpg',
      genre_ids: [28, 878, 12],
      id: 565770,
      original_language: 'en',
      original_title: 'Blue Beetle',
      overview:
        'Recent college grad Jaime Reyes returns home full of aspirations for his future, only to find that home is not quite as he left it. As he searches to find his purpose in the world, fate intervenes when Jaime unexpectedly finds himself in possession of an ancient relic of alien biotechnology: the Scarab.',
      popularity: 579.577,
      poster_path: '/mXLOHHc1Zeuwsl4xYKjKh2280oL.jpg',
      release_date: '2023-08-16',
      title: 'Blue Beetle',
      video: false,
      vote_average: 7,
      vote_count: 1612,
    },
    {
      adult: false,
      backdrop_path: '/pA3vdhadJPxF5GA1uo8OPTiNQDT.jpg',
      genre_ids: [28, 18],
      id: 678512,
      original_language: 'en',
      original_title: 'Sound of Freedom',
      overview:
        'The story of Tim Ballard, a former US government agent, who quits his job in order to devote his life to rescuing children from global sex traffickers.',
      popularity: 614.334,
      poster_path: '/qA5kPYZA7FkVvqcEfJRoOy4kpHg.jpg',
      release_date: '2023-07-03',
      title: 'Sound of Freedom',
      video: false,
      vote_average: 8.1,
      vote_count: 1459,
    },
    {
      adult: false,
      backdrop_path: '/dZbLqRjjiiNCpTYzhzL2NMvz4J0.jpg',
      genre_ids: [27, 53],
      id: 951491,
      original_language: 'en',
      original_title: 'Saw X',
      overview:
        "Between the events of 'Saw' and 'Saw II', a sick and desperate John Kramer travels to Mexico for a risky and experimental medical procedure in hopes of a miracle cure for his cancer, only to discover the entire operation is a scam to defraud the most vulnerable. Armed with a newfound purpose, the infamous serial killer returns to his work, turning the tables on the con artists in his signature visceral way through devious, deranged, and ingenious traps.",
      popularity: 626.145,
      poster_path: '/aQPeznSu7XDTrrdCtT5eLiu52Yu.jpg',
      release_date: '2023-09-26',
      title: 'Saw X',
      video: false,
      vote_average: 7.4,
      vote_count: 1065,
    },
    {
      adult: false,
      backdrop_path: '/agJTpYOjSSG9uZe5xVZHL8nE6nj.jpg',
      genre_ids: [27, 9648, 53],
      id: 841742,
      original_language: 'fr',
      original_title: 'Megalomaniac',
      overview:
        "Felix and Martha, the two offspring of legendary serial murderer The Skinner of Mons, grapple with the grotesque legacy bequeathed to them. While Martha works a menial janitorial job, Felix continues his father's reign of terror. But following a brutal attack at work, Martha quickly descends into madness, finding she must come to terms with the infamous blood that flows through her veins.",
      popularity: 502.741,
      poster_path: '/eAx5QKnboZpysIg1XBfjhbSgOPF.jpg',
      release_date: '2023-09-08',
      title: 'Megalomaniac',
      video: false,
      vote_average: 6.2,
      vote_count: 13,
    },
    {
      adult: false,
      backdrop_path: '/mRmRE4RknbL7qKALWQDz64hWKPa.jpg',
      genre_ids: [80, 53],
      id: 800158,
      original_language: 'en',
      original_title: 'The Killer',
      overview:
        "After a fateful near-miss, an assassin battles his employers, and himself, on an international manhunt he insists isn't personal.",
      popularity: 447.212,
      poster_path: '/e7Jvsry47JJQruuezjU2X1Z6J77.jpg',
      release_date: '2023-10-25',
      title: 'The Killer',
      video: false,
      vote_average: 6.8,
      vote_count: 736,
    },
    {
      adult: false,
      backdrop_path: '/AprNYUAS2AJ3xVgg7Wwt00GVvsM.jpg',
      genre_ids: [16, 10751, 28, 878],
      id: 893723,
      original_language: 'en',
      original_title: 'PAW Patrol: The Mighty Movie',
      overview:
        'A magical meteor crash lands in Adventure City and gives the PAW Patrol pups superpowers, transforming them into The Mighty Pups.',
      popularity: 540.935,
      poster_path: '/oJkyw6TNEWIS0WkrxNspJfUTao4.jpg',
      release_date: '2023-09-21',
      title: 'PAW Patrol: The Mighty Movie',
      video: false,
      vote_average: 6.9,
      vote_count: 172,
    },
    {
      adult: false,
      backdrop_path: '/5mzr6JZbrqnqD8rCEvPhuCE5Fw2.jpg',
      genre_ids: [28, 878, 27],
      id: 615656,
      original_language: 'en',
      original_title: 'Meg 2: The Trench',
      overview:
        'An exploratory dive into the deepest depths of the ocean of a daring research team spirals into chaos when a malevolent mining operation threatens their mission and forces them into a high-stakes battle for survival.',
      popularity: 544.8,
      poster_path: '/4m1Au3YkjqsxF8iwQy0fPYSxE0h.jpg',
      release_date: '2023-08-02',
      title: 'Meg 2: The Trench',
      video: false,
      vote_average: 6.8,
      vote_count: 2544,
    },
    {
      adult: false,
      backdrop_path: '/dRWhJ4godwy40JdmNuRZy23oViY.jpg',
      genre_ids: [27],
      id: 830764,
      original_language: 'en',
      original_title: 'Pet Sematary: Bloodlines',
      overview:
        'In 1969, a young Jud Crandall has dreams of leaving his hometown of Ludlow, Maine behind, but soon discovers sinister secrets buried within and is forced to confront a dark family history that will forever keep him connected to Ludlow.',
      popularity: 553.407,
      poster_path: '/yqnNLn24shYnZ6kqGpbwuB3NJ0D.jpg',
      release_date: '2023-09-23',
      title: 'Pet Sematary: Bloodlines',
      video: false,
      vote_average: 6.1,
      vote_count: 201,
    },
    {
      adult: false,
      backdrop_path: '/r7DuyYJ0N3cD8bRKsR5Ygq2P7oa.jpg',
      genre_ids: [12, 28, 18],
      id: 980489,
      original_language: 'en',
      original_title: 'Gran Turismo',
      overview:
        'The ultimate wish-fulfillment tale of a teenage Gran Turismo player whose gaming skills won him a series of Nissan competitions to become an actual professional racecar driver.',
      popularity: 556.183,
      poster_path: '/51tqzRtKMMZEYUpSYkrUE7v9ehm.jpg',
      release_date: '2023-08-09',
      title: 'Gran Turismo',
      video: false,
      vote_average: 8,
      vote_count: 1445,
    },
    {
      adult: false,
      backdrop_path: '/9WxqnP9c29wXd03sALSpxpTx8fk.jpg',
      genre_ids: [9648, 53, 28],
      id: 1172009,
      original_language: 'en',
      original_title: 'The Black Book',
      overview:
        'After his son is wrongly accused of kidnapping, a deacon who has just lost his wife takes matters into his own hands and fights a crooked police gang to clear him.',
      popularity: 423.844,
      poster_path: '/kn28W24slBLyGr8ZIZnxNE5YZrY.jpg',
      release_date: '2023-09-22',
      title: 'The Black Book',
      video: false,
      vote_average: 6.5,
      vote_count: 57,
    },
    {
      adult: false,
      backdrop_path: '/pLm9j7o5InoWaG2tlaQABYR2cAx.jpg',
      genre_ids: [27, 9648, 53],
      id: 968051,
      original_language: 'en',
      original_title: 'The Nun II',
      overview:
        'In 1956 France, a priest is violently murdered, and Sister Irene begins to investigate. She once again comes face-to-face with a powerful evil.',
      popularity: 520.807,
      poster_path: '/5gzzkR7y3hnY8AD1wXjCnVlHba5.jpg',
      release_date: '2023-09-06',
      title: 'The Nun II',
      video: false,
      vote_average: 6.9,
      vote_count: 1422,
    },
  ],
  total_pages: 41071,
  total_results: 821408,
};

export const moviePopularList: IMovieState = {
  loading: false,
  data: movieItem2,
  error: null,
};
