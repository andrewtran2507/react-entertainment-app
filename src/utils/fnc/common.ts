import { IErrorResponse, TErrors, defElm } from 'utils/constant';

export const getImageULR = (
  path: string | null | undefined,
  folder: string,
): string =>
  path
    ? `${process.env.REACT_APP_IMAGE_URL}/${folder}${path}`
    : '/assets/images/no-img.png';

export const mockMatchMedia = (): void => {
  window.matchMedia = (query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  });
};

export const getErrorResponse = (error: TErrors): IErrorResponse => {
  let objError: IErrorResponse = {
    status_code: 2,
    status_message: 'Something when wrong!',
    success: false,
  };
  if (!error) {
    return objError;
  }
  if (error.response.data) {
    objError = error?.response?.data;
  }
  if (error?.name === 'ZodError') {
    objError.status_message =
      error?.errors
        ?.map(
          (d: { path: string[]; message: string }) =>
            `${d?.path[d?.path.length - 1]} ${d?.message}`,
        )
        .join(', ') || objError.status_message;
  }
  return objError;
};

export const sleep = (ms: number) =>
  new Promise((resolve) => setTimeout(resolve, ms));

export const elmQuerySelector = (
  elm: Element,
  queryStr: string,
  isAll = false,
) =>
  (isAll
    ? Array.from(elm.querySelectorAll(queryStr))
    : elm.querySelector(queryStr)) || defElm;
