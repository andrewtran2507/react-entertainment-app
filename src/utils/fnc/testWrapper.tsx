import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { movieReducer } from 'pages/Home/Redux';
import { movieDetailReducer } from 'pages/MovieDetail/Redux';
import MainLayout from 'components/Layout/MainLayout';
import { IMovieState } from 'pages/Home/Redux/types';
import { IMovieDetailState } from 'pages/MovieDetail/Redux/types';
import { ToolkitStore } from '@reduxjs/toolkit/dist/configureStore';
const defStore = {
  reducer: { movie: movieReducer, movieDetail: movieDetailReducer },
};

function render(
  ui: ReactElement,
  {
    preloadedState = {},
    store = configureStore({ ...defStore, preloadedState }),
    ...renderOptions
  }: {
    preloadedState?: object;
    store?: ToolkitStore<{
      movie: IMovieState;
      movieDetail: IMovieDetailState;
    }>;
  },
) {
  function Wrapper({ children }: { children: ReactElement }) {
    return (
      <Provider store={store}>
        <BrowserRouter>
          <MainLayout>{children}</MainLayout>
        </BrowserRouter>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };
