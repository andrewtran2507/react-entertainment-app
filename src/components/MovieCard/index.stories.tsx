import { Meta, StoryObj } from '@storybook/react';
import {
  reactRouterParameters,
  withRouter,
} from 'storybook-addon-react-router-v6';
import MovieCard from './index';
import { movieItem } from './mock';

const meta: Meta<typeof MovieCard> = {
  title: 'components/MovieCard',
  component: MovieCard,
  decorators: [withRouter],
  parameters: {
    layout: 'fullscreen',
    reactRouter: reactRouterParameters({
      location: {
        pathParams: { movieId: 385687 },
      },
      routing: { path: '/movie/:movieId' },
    }),
  },
  tags: ['autodocs'],
  argTypes: {
    movieItem: { control: 'object' },
  },
};

export default meta;

type Story = StoryObj<typeof MovieCard>;

export const OnLoading: Story = {
  args: {
    movieItem: null,
  },
};

export const UnLoading: Story = {
  args: {
    movieItem: movieItem,
  },
};
