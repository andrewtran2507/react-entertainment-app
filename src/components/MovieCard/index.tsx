import React, { FC, useEffect, useState } from 'react';
import { Card, Progress, Skeleton, Typography } from 'antd';
import { useNavigate } from 'react-router-dom';
import { LazyLoadImage } from 'react-lazy-load-image-component';
import { ZMovieItemModel } from 'pages/Home/Redux/types';
import { conicColors } from 'utils/css/colors';
import { getImageULR } from 'utils/fnc/common';
import { imageFolder } from 'utils/constant';
import './index.scss';

const { Meta } = Card;
const { Paragraph } = Typography;

const MovieCard: FC<{
  movieItem: ZMovieItemModel | null;
  indx: number;
}> = ({ movieItem, indx = 1 }) => {
  const [cLoading, setCLoading] = useState(true);
  const covertImage = getImageULR(
    movieItem?.poster_path,
    imageFolder['220x330'],
  );
  const navigate = useNavigate();

  const movieDetailURL = `/movie/${movieItem?.id}`;

  useEffect(() => {
    const tm = setTimeout(
      () => {
        setCLoading(false);
      },
      50 * (indx + 1),
    );

    return () => {
      clearTimeout(tm);
    };
  }, [cLoading]);

  return (
    <div className="movie-card">
      {cLoading || !movieItem ? (
        <div className="skeleton-movie-card">
          <div className="skeleton-img">
            <Skeleton.Image active />
          </div>
          <Skeleton avatar active />
        </div>
      ) : (
        <Card
          hoverable
          cover={
            <LazyLoadImage
              alt="cover-image"
              effect="blur"
              wrapperProps={{
                style: { transitionDelay: '1s' },
              }}
              src={covertImage}
            />
          }
          onClick={() => {
            navigate(movieDetailURL);
          }}
        >
          <Meta
            avatar={
              <Progress
                type="circle"
                percent={movieItem?.vote_average * 10}
                size={48}
                strokeColor={conicColors}
              />
            }
            title={
              <a href={movieDetailURL} target="_self">
                {movieItem?.title || 'N/A'}
              </a>
            }
            description={
              <Paragraph ellipsis>{movieItem?.overview || 'N/A'}</Paragraph>
            }
          />
        </Card>
      )}
    </div>
  );
};

export default MovieCard;
