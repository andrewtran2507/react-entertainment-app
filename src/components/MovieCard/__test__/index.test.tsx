import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import {
  cleanup,
  render,
  waitFor,
  screen,
  RenderResult,
  act,
} from '@testing-library/react';
import '@testing-library/jest-dom';

import { movieItem } from '../mock';
import MovieCard from 'components/MovieCard';
import { mockMatchMedia } from 'utils/fnc/common';

describe('Test MovieCard Component', () => {
  let component: RenderResult | undefined = undefined;
  beforeAll(async () => {
    mockMatchMedia();
  });

  afterEach(() => {
    cleanup();
  });

  it('Should match snapshot of MovieCard component', async () => {
    await act(() => {
      component = render(
        <BrowserRouter>
          <MovieCard movieItem={movieItem} indx={1} />
        </BrowserRouter>,
      );
    });
    const { asFragment } = component || { asFragment: () => DocumentFragment };
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test content on MovieCard Component', async () => {
    render(
      <BrowserRouter>
        <MovieCard movieItem={movieItem} indx={1} />
      </BrowserRouter>,
    );
    await waitFor(() =>
      expect(screen.getByText(/Fast X/i)).toBeInTheDocument(),
    );
  });
});
