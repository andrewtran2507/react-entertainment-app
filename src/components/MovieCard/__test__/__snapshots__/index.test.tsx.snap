// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MovieCard Component Should match snapshot of MovieCard component 1`] = `
<DocumentFragment>
  <div
    class="movie-card"
  >
    <div
      class="skeleton-movie-card"
    >
      <div
        class="skeleton-img"
      >
        <div
          class="ant-skeleton ant-skeleton-element ant-skeleton-active css-dev-only-do-not-override-2i2tap"
        >
          <div
            class="ant-skeleton-image"
          >
            <svg
              class="ant-skeleton-image-svg"
              viewBox="0 0 1098 1024"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="ant-skeleton-image-path"
                d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="ant-skeleton ant-skeleton-with-avatar ant-skeleton-active css-dev-only-do-not-override-2i2tap"
      >
        <div
          class="ant-skeleton-header"
        >
          <span
            class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
          />
        </div>
        <div
          class="ant-skeleton-content"
        >
          <h3
            class="ant-skeleton-title"
            style="width: 50%;"
          />
          <ul
            class="ant-skeleton-paragraph"
          >
            <li />
            <li />
          </ul>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
